ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -252,9 +254,57 @@ private[kvutils] class TransactionCommitter(
   private def blind: Step =
     (commitContext, transactionEntry) => {
       val blindingInfo = Blinding.blind(transactionEntry.transaction)
-      buildFinalResult(commitContext, transactionEntry, blindingInfo)
+      buildFinalResultState(commitContext, transactionEntry, blindingInfo)
     }
 
+  private def removeUnnecessaryNodes: Step = (commitContext, transactionEntry) => {
+    val transaction = transactionEntry.submission.getTransaction
+    val nodes: Map[String, TransactionOuterClass.Node] =
+      transaction.getNodesList.asScala.map(n => n.getNodeId -> n)(scala.collection.breakOut)
+    val nodesToKeep: Set[String] = nodes.collect {
+      case (id, node) if node.hasCreate || node.hasExercise => id
+    }(scala.collection.breakOut)
```

Comment:
```suggestion
    val nodesToKeep: Set[String] = nodes.iterator.collect {
      case (id, node) if node.hasCreate || node.hasExercise => id
    }.toSet
```
Same point as above.